Project1

In [1]:
#import sequential library pyhton keras libaray
#the sequential model is for creating model layer by layer 
from keras.models import Sequential
#import dense layers fuction for creating neurons in layer
from keras.layers import Dense
#import numpy library to add high-level mathematical functions to operate on datas  
import numpy
#The random module uses the seed value as a base to generate a random number.
#Seeding a pseudo-random number generator gives it its first "previous" value.
#That is, if you have the same seed, you will get the same sequence of numbers twice.
numpy.random.seed(7)
#load the csv, split string with ","
dataset = numpy.loadtxt("pima-indians-diabetes.csv", delimiter=",")
#assigned columns one to eight to x as input
#assigned column nine to y as output
X = dataset[:,0:8]
Y = dataset[:,8]
type(X)

/Users/lily/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.


numpy.ndarray

In [2]:
#define a sequence of layers
model = Sequential()
#8 input parameters, with 12 neurons in the FIRST hidden layer, the Rectified Linear Unit- Relu
model.add(Dense(12, input_dim=8, activation='relu'))
#8 neurons in the second hidden layer, the Rectified Linear Unit- Relu
model.add(Dense(8, activation='relu'))
#the output 1/0
model.add(Dense(1, activation='sigmoid'))

In [3]:
#complie model
#loss means cross entropy between the network output and target
#Adam Optimization Algorithm 
#A metric is a function that is used to judge the performance of your model
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

In [4]:
#fit the profermance of model
#The number of epochs is a hyperparameter that defines the number times that the learning algorithm will work through the entire dataset.
#The batch size is a hyperparameter that defines the number of samples to work through before updating the internal model parameters.
model.fit(X, Y, epochs=150, batch_size=10)

Epoch 1/150
768/768 [==============================] - 0s 436us/step - loss: 3.7110 - acc: 0.5990
Epoch 2/150
768/768 [==============================] - 0s 95us/step - loss: 0.9366 - acc: 0.5924
Epoch 3/150
768/768 [==============================] - 0s 97us/step - loss: 0.7472 - acc: 0.6406
Epoch 4/150
768/768 [==============================] - 0s 99us/step - loss: 0.7115 - acc: 0.6576
Epoch 5/150
768/768 [==============================] - 0s 97us/step - loss: 0.6823 - acc: 0.6745
Epoch 6/150
768/768 [==============================] - 0s 100us/step - loss: 0.6511 - acc: 0.6849
Epoch 7/150
768/768 [==============================] - 0s 95us/step - loss: 0.6495 - acc: 0.6758
Epoch 8/150
768/768 [==============================] - 0s 96us/step - loss: 0.6369 - acc: 0.6836
Epoch 9/150
768/768 [==============================] - 0s 95us/step - loss: 0.6246 - acc: 0.6979
Epoch 10/150
768/768 [==============================] - 0s 97us/step - loss: 0.6293 - acc: 0.6771
Epoch 11/150
768/768 [=====

In [5]:
#evaluate the profermance of model
#The average and standard deviation of the model performance is then printed at the end of the run to provide a robust estimate of model accuracy.
scores = model.evaluate(X, Y)
print("\n%s: %.2f%%" % (model.metrics_names[1], scores[1]*100))

768/768 [==============================] - 0s 56us/step

acc: 79.56%


Project2

In [1]:
import numpy as np
# Import pandas for using common data science methods
import pandas as pd
# Import train_test_split from sklearn library
# Read the dataset from csv fiel using pandas library
from sklearn.model_selection import train_test_split
from keras.models import Sequential
#import dense layers fuction for creating neurons in layer
from keras.layers import Dense 
dataset = pd.read_csv('train_validation.csv',encoding = "ISO-8859-1")
Y=dataset.iloc[:,0]
X=dataset.iloc[:,1]
Y = Y.as_matrix(columns=None)



/Users/lily/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
/Users/lily/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:13: FutureWarning: Method .as_matrix will be removed in a future version. Use .values instead.
  del sys.path[0]


In [2]:
#preprocessing data
from nltk.corpus import stopwords
stopwo = stopwords.words('english')
test = pd.DataFrame(X)
test.columns = ["tweet"]
test['tweet_without_stopwords'] = test['tweet'].apply(lambda x: ' '.join([word for word in x.split() if word not in (stopwo)]))
print(test['tweet_without_stopwords'])
X=test['tweet_without_stopwords']


0                              sad APL friend.............
1                             I missed New Moon trailer...
2                                      omg already 7:30 :O
3        .. Omgaga. Im sooo im gunna CRy. I've dentist ...
4                           think mi bf cheating me!!! T_T
5                                              worry much?
6                       Juuuuuuuuuuuuuuuuussssst Chillin!!
7                 Sunny Again Work Tomorrow :-| TV Tonight
8                      handed uniform today . miss already
9                              hmmmm.... wonder number @-)
10                                 I must think positive..
11                         thanks haters face day! 112-102
12                                      weekend sucked far
13                         jb isnt showing australia more!
14                                           ok thats win.
15                 &lt;-------- This way feel right now...
16       awhhe man.... I'm completely useless rt now. F.

In [3]:
#convert text to features
from sklearn.feature_extraction.text import HashingVectorizer
vectorizer = HashingVectorizer(n_features=1000)
# encode document
vector = vectorizer.transform(X)
# summarize encoded vector
print(vector.shape)
print(vector.toarray())


(99989, 1000)
[[0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 ...
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]
 [0. 0. 0. ... 0. 0. 0.]]


In [4]:
X=vector.toarray()

In [5]:
#split traing data and test data by the proportion of 8:2 
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.2, random_state = 2)


In [6]:
#define a sequence of layers
model = Sequential()
#there are four layers in the model, 1000 features
model.add(Dense(12, input_dim=1000, activation='relu'))
model.add(Dense(12, activation='relu'))
model.add(Dense(12, activation='relu'))
model.add(Dense(12, activation='relu'))
#the output 1/0
model.add(Dense(1, activation='sigmoid'))

In [7]:
#complie model
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

In [8]:
#fit the profermance of model
model.fit(X_train, Y_train, epochs=50, batch_size=10)

Epoch 1/50
79991/79991 [==============================] - 11s 136us/step - loss: 0.5976 - acc: 0.6713
Epoch 2/50
79991/79991 [==============================] - 10s 123us/step - loss: 0.5644 - acc: 0.7035
Epoch 3/50
79991/79991 [==============================] - 10s 121us/step - loss: 0.5478 - acc: 0.7195
Epoch 4/50
79991/79991 [==============================] - 10s 122us/step - loss: 0.5330 - acc: 0.7337
Epoch 5/50
79991/79991 [==============================] - 10s 122us/step - loss: 0.5199 - acc: 0.7461
Epoch 6/50
79991/79991 [==============================] - 10s 121us/step - loss: 0.5084 - acc: 0.7563
Epoch 7/50
79991/79991 [==============================] - 10s 123us/step - loss: 0.4992 - acc: 0.7654
Epoch 8/50
79991/79991 [==============================] - 10s 122us/step - loss: 0.4908 - acc: 0.7717
Epoch 9/50
79991/79991 [==============================] - 10s 122us/step - loss: 0.4832 - acc: 0.7792
Epoch 10/50
79991/79991 [==============================] - 10s 122us/step - loss: 

In [9]:
#the accuracy of train data
scores = model.evaluate(X_train, Y_train)
print("\n%s: %.2f%%" % (model.metrics_names[1], scores[1]*100))

79991/79991 [==============================] - 1s 16us/step

acc: 85.34%


In [11]:
#the accuracy of test data
scores = model.evaluate(X_test, Y_test)
print("\n%s: %.2f%%" % (model.metrics_names[1], scores[1]*100))

19989/19989 [==============================] - 0s 22us/step

acc: 81.00%


First, I used numpy and pandas to upload csv file into jupter notebook.
Second, I used NLTK to remove "stopwords" from sentiment text. 
Third, I transfer words into 500 features.
Forth, I built a binary classification model with 500 inputs, 4 hidden layers and each layers have 12 neurals. the activation is rectified linear unit(ReLU) 
The output is limited to 0 or 1.

the accuracy of train data is 85.34%
the accuracy of test data is 81.00%